frontend/src/components/UserDetail/DriverDetail.tsx

```diff
@@ -17,23 +33,89 @@ type DriverDetailProps = {
 const DriverDetail = () => {
   const location = useLocation<DriverDetailProps>();
   const driver: DriverDetailProps = location.state;
-  const availToString = (acc: string, [day, timeRange]: string[]) => acc + day + ": " + timeRange + " • ";
-  const parsedAvail = driver.availability.reduce(availToString, "");
+  const availToString = (acc: string, [day, timeRange]: string[]) => acc + day + ': ' + timeRange + ' • ';
+  const parsedAvail = driver.availability.reduce(availToString, '');
   const avail = parsedAvail.substring(0, parsedAvail.length - 2);
-  const vehicle = driver.vehicle ? (driver.vehicle.name + " (" + driver.vehicle.capacity + " people)") : ""
+  const vehicle = driver.vehicle ? (driver.vehicle.name + ' (' + driver.vehicle.capacity + ' people)') : '';
+  const [rides, setRides] = useState<Ride[]>([]);
+
+  const compRides = (a: Ride, b: Ride) => {
+    const x = new Date(a.startTime);
+    const y = new Date(b.startTime);
+    if (x < y) return -1;
+    if (x > y) return 1;
+    return 0;
+  };
+
+  const getPastRides = () => {
+    fetch(`/rides?type=past&driver=${driver.id}`)
+      .then((res) => res.json())
+      .then(({ data }) => setRides(data.sort(compRides)));
+  };
+
+  useEffect(getPastRides, []);
+
+  function renderTableData(allRides: Ride[]) {
+    return allRides.map((ride, index) => {
+      const date = new Date(ride.startTime).toLocaleDateString();
+      const { rider } = ride;
+      const name = rider ? `${rider.firstName} ${rider.lastName}` : '';
+      const needs = rider ? (rider.accessibilityNeeds || []).join(', ') : '';
```

Comment:
I used `` rider ? `${rider.firstName} ${rider.lastName}` : '' `` before because I was getting unscheduled rides. The fetch to past rides should ensure existence of ride and ride.startLocation.

If past rides are expected to have all fields filled in, maybe just do `` `${rider.firstName} ${rider.lastName}` `` (same for other fields).

Reply:
The problem is some of the existing data in the dev stage has invalid data where some don't have a start location or a name so I included those checks.